### DISCUSSION TITLE
React (WebComponent) nested swipers with virtual slides not working

### DISCUSSION THREAD

**Discussion by kirgans** [nolimits4web/swiper#6859] [Q&A]
Im trying to use swiper.js with react (webComponent) with nested swipers and virtual slides, but can't get this to work :/ 
You can check the attached codesandbox, if you comment `virtual={true}` on both swipers this is working great but without **virtual slides** and as Im working with lot of slides this can 't be an option.

https://codesandbox.io/p/sandbox/nest-swiper-virtual-slides-q59yq6?file=%2Fapp%2Fpage.tsx%3A26%2C31

`"swiper": "^10.0.4"`

Any help would be appreciated 😄 
Thanks.

**Comment by kirgans** [comment]
any hint on this @nolimits4web ? Iv'e seen this bug in other versions, I don't know if this is even possible ?  thanks for your help

**Comment by nolimits4web** [comment]
You need to pass slides as virtual.slides data in parent swiper like so:
```html
<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8" />
  <title>Swiper Playground</title>
  <meta name="viewport" content="width=device-width" />
</head>

<body>
  <swiper-container init="false"></swiper-container>
  <script type="module">
    import { register } from 'swiper/element/bundle';
    register();

    document.querySelector('swiper-container').direction = 'vertical';

    document.querySelector('swiper-container').virtual = {
      enabled: true,
      cache: false,
      slides: Array.from({ length: 10 }).map(() => `
      <swiper-slide>
        <swiper-container virtual>
          <swiper-slide>H Slide 1</swiper-slide>
          <swiper-slide>H Slide 2</swiper-slide>
          <swiper-slide>H Slide 3</swiper-slide>
          <swiper-slide>H Slide 4</swiper-slide>
          <swiper-slide>H Slide 5</swiper-slide>
          <swiper-slide>H Slide 6</swiper-slide>
          <swiper-slide>H Slide 7</swiper-slide>
          <swiper-slide>H Slide 8</swiper-slide>
          <swiper-slide>H Slide 9</swiper-slide>
          <swiper-slide>H Slide 10</swiper-slide>
        </swiper-container>
      </swiper-slide>
      `)
    }
    document.querySelector('swiper-container').initialize()
  </script>
</body>

</html>
```

**Reply by kirgans** [reply]
Thanks for your quick answer, however I can't get this to work properly.
In addition, I'm trying to get this work in a react way using WebComponent using this article as support (https://swiperjs.com/blog/using-swiper-element-in-react).
Can you provide a working sample please or hints working with react and web component as Swiper React components will likely to be removed in future versions?

**Reply by kirgans** [reply]
I managed to get something working with your code, BUT as soon as a virtual slide which was not rendered is loading (from the parent swiper) children swiper is empty. Still can't get this work in a react/web component way.

**Reply by nolimits4web** [reply]
Do you have a codesandbox of "get something working with your code"?

**Reply by kirgans** [reply]
sure https://codesandbox.io/p/sandbox/flamboyant-yalow-gx7wkw?file=%2Fapp%2Fpage.tsx%3A19%2C26
(don't know why there is trailing commas, nvm)

**Reply by kirgans** [reply]
Ok, somehow, I managed to get the nested swipers with virtual slides working in a react way and with web components :
https://codesandbox.io/p/sandbox/nolimits4web-swiper-discussions-6859-forked-h3pmf6 

But now when you going down to the vertical slides and then go up, horizontal slides start disappearing 😳
it's reproductible in the attached codesandbox.

Should I open a new discussion or we can keep this one for this new issue ?

thank you in advance 🙌

**Reply by kirgans** [reply]
I had some tests this morning, only happens with virtual slides...

**Reply by kirgans** [reply]
@nolimits4web  any idea ?

**Reply by nolimits4web** [reply]
I've updated blog post about how add virtual slides in react-way too https://swiperjs.com/blog/using-swiper-element-in-react#virtual-slides

**Reply by kirgans** [reply]
Thanks for your reply and for the blog post @nolimits4web .
Maybe I'm missing something but I can't get it works following the blog post :

`Error: React.cloneElement(...): The argument must be a React element, but you passed undefined.`
` return cloneElement(slide, {...` --> slide is undefined

inside renderExternal, vd.slides are undefined

**Reply by kirgans** [reply]
```
        <swiper-container
            ref={ref}
            virtual="true"
            style={...style}
        >
            {children}
        </swiper-container>
```
This is working, but as soon as I do like in the blog post, it's not working for virtual element,
can you provide a working example maybe ? 

In addition, loop={true} doesn't work if slidesPerView works with decimals, like slidesPerView={4.5},  is it on purpose ?
(total number of slides is >= slidesPerView * 2)

edit: it's working with slidesPerView={4} but not with 6 on @1.33 device on my case, looks like it depends on the resolution or elements width

Thanks

**Reply by kirgans** [reply]
any update @nolimits4web  ?
